### PR TITLE
Add capability to use different types for enums on protocol vs convenience

### PR DESCRIPTION
**Pull request by m-nash**
Fixes https://github.com/microsoft/typespec/issues/3953

### CODE REVIEW

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ClientProviderTests.cs [hunk 1]
```diff
@@ -334,6 +334,115 @@ public void TestSubClientAccessorFactoryMethods(InputClient client, bool hasSubC
...
+            MockHelpers.LoadMockPlugin();
+
+            //protocol and convenience methods should have a different type for enum query parameters
+            var clientProvider = ClientModelPlugin.Instance.TypeFactory.CreateClient(GetEnumQueryParamClient());
+            Assert.IsNotNull(clientProvider);
+            var methods = clientProvider.Methods;
+            //4 methods, sync / async + protocol / convenience
+            Assert.AreEqual(4, methods.Count);
+            //two methods need to have the query parameter as an enum
+            var convenienceMethods = methods.Where(m => m.Signature.Parameters.Any(p => p.Name == "queryParam" && p.Type.Name == "InputEnum"));
```
Comment (JoshLove-msft) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ClientProviderTests.cs:349]: Should we use the variable?

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/TypeFactory.cs [hunk 2]
```diff
@@ -220,14 +217,7 @@ internal CSharpType CreatePrimitiveCSharpType(InputType inputType)
         /// <param name="parameter">The <see cref="InputParameter"/> to convert.</param>
         /// <returns>An instance of <see cref="ParameterProvider"/>.</returns>
         public virtual ParameterProvider CreateParameter(InputParameter parameter)
```
Comment (JoshLove-msft) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/TypeFactory.cs:219]: Should we have the protected CreateParameterCore method and make the public method not be virtual?

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ParameterProvider.cs [hunk 3]
```diff
@@ -28,7 +28,7 @@ public sealed class ParameterProvider : IEquatable<ParameterProvider>
         public ParameterValidationType Validation { get; init; } = ParameterValidationType.None;
         public bool IsRef { get; }
         public bool IsOut { get; }
-        internal IReadOnlyList<AttributeStatement> Attributes { get; } = Array.Empty<AttributeStatement>();
+        public IReadOnlyList<AttributeStatement> Attributes { get; } = [];
```
Comment (JoshLove-msft) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ParameterProvider.cs:31]: Why is this public?

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/TypeFactory.cs (hunk 2 above)
Comment (JoshLove-msft) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/TypeFactory.cs:219]: Nevermind - on the protected method, but this should probably not be virtual.

**Review by JoshLove-msft — APPROVED** [review]

File: packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ClientProviderTests.cs [hunk 4]
```diff
@@ -334,6 +334,115 @@ public void TestSubClientAccessorFactoryMethods(InputClient client, bool hasSubC
...
+        private class ValidateQueryParamDiffClientProvider : ClientProvider
+        {
+            public ValidateQueryParamDiffClientProvider(InputClient client)
+                : base(client)
+            {
+            }
+
+            protected override MethodProvider[] BuildMethods()
+            {
+                var method = base.BuildMethods().Where(m => m.Signature.Parameters.Any(p => p.Name == "queryParam" && p.Type.Name == "InputEnum" && !m.Signature.Name.EndsWith("Async"))).First();
```
Comment (JoshLove-msft) [inline packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ClientProviderTests.cs:436]: Could we have both the convenience and protocol methods?

Comment (m-nash) [reply packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/test/Providers/ClientProviders/ClientProviderTests.cs:435]: we want to validate the call from conv to protocol so having both does not make sense in this context.

**Pull request merged**